src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -257,31 +268,32 @@ private void HandleRemovedDocuments(RequestContext context, DiagnosticParams[] p
             }
         }
 
+        /// <summary>
+        /// Determines if diagnostics have changed since the last report based on the request resultId and
+        /// the resultId stored in <see cref="_documentIdToLastResultId"/>  Called under <see cref="_gate"/>
+        /// </summary>
         private bool DiagnosticsAreUnchanged(Dictionary<Document, DiagnosticParams> documentToPreviousDiagnosticParams, Document document)
```

Comment:
consider moving this as a local function into the caller.  to make it clearer it's held by a lock.